From 52f21750b51c330e9d6baef00ad86a0a200fbc11 Mon Sep 17 00:00:00 2001
From: Tim Hartley <timothy.d.hartley@intel.com>
Date: Mon, 14 Oct 2013 21:49:53 -0700
Subject: Dalvik: PCG GL assert build broken

BZ: 145137

This patch fixes a trivial problem to allow building with
asserts enabled.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ic2f88b321ccde702f9e231e23d1d8477369c6c93
Orig-MCG-Change-Id: If50121649a34296abe6ce1b4d842d1cd0c88c555
Signed-off-by: Tim Hartley <timothy.d.hartley@intel.com>
Reviewed-on: http://android.intel.com:8080/138449
Reviewed-by: Kreitzer, David L <david.l.kreitzer@intel.com>
Reviewed-by: Banerji, Udayan <udayan.banerji@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: Dittert, Eric <eric.dittert@intel.com>
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/pcg/CompilationUnitPCG.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/codegen/x86/pcg/CompilationUnitPCG.cpp b/vm/compiler/codegen/x86/pcg/CompilationUnitPCG.cpp
index 531de22..7ff2afe 100644
--- a/vm/compiler/codegen/x86/pcg/CompilationUnitPCG.cpp
+++ b/vm/compiler/codegen/x86/pcg/CompilationUnitPCG.cpp
@@ -128,7 +128,7 @@ CGTemp CompilationUnitPCG::getCGTempForXMM (int xmmNum)
 {
     // We reserved the first gDvmJitGlobals.vectorRegisters available CGTemp
     // numbers after the max SSA number for the XMM registers.
-    assert (xmmNum >= 0 && xmmNum < gDvmJitGlobals.vectorRegisters);
+    assert (xmmNum >= 0 && xmmNum < gDvmJit.vectorRegisters);
 
     return numSSARegs + xmmNum;
 }
-- 
1.7.4.1

